lib/Doctrine/ORM/Internal/Hydration/SimpleObjectHydrator.php

```diff
@@ -124,11 +124,11 @@ protected function hydrateRowData(array $sqlResult, array &$result)
             }
         }
 
-        if (isset($this->hints[Query::HINT_REFRESH_ENTITY])) {
-            $id = $this->em->getIdentifierFlattener()->flattenIdentifier($this->class, $data);
-
-            $this->em->getUnitOfWork()->registerManaged($this->hints[Query::HINT_REFRESH_ENTITY], $id, $data);
-        }
+//        if (isset($this->hints[Query::HINT_REFRESH_ENTITY])) {
```

Comment:
This code was extremely broken, but I still need to see if any other related code may have been broken by it.